server_addon/houdini/client/ayon_houdini/plugins/publish/collect_files_for_cleaning_up.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+import clique
+import pyblish.api
+from ayon_core.pipeline import AYONPyblishPluginMixin
+from ayon_houdini.api import plugin
+
+
+class CollectFilesForCleaningUp(plugin.HoudiniInstancePlugin,
+                                AYONPyblishPluginMixin):
+    """Collect Files For Cleaning Up.
+
+    This collector collects output files
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector registers exported files and
+          the parent folder (if it was empty) for deletion
+          in `ExplicitCleanUp` plug-in.
+        Artists are free to change the file path in the ROP node.
+    """
+
+    # It should run after CollectFrames and Collect Render plugins,
+    # and before CollectLocalRenderInstances.
+    order = pyblish.api.CollectorOrder + 0.115
+
+    hosts = ["houdini"]
+    families = ["*"]
+    label = "Collect Files For Cleaning Up"
+    intermediate_exported_render = False
+
+    def process(self, instance):
+
+        files = []
+        staging_dirs = []
+
+        expected_files = instance.data.get("expectedFiles", [])
+
+        # 'expectedFiles' are preferred over 'frames'
+        if expected_files:
+            # Products with expected files
+            # This can be Render products or submitted cache to farm.
+            for expected in expected_files:
+                # expected.values() is a list of lists
+                for output_files in expected.values():
+                    staging_dir, _ = os.path.split(output_files[0])
+                    if staging_dir not in staging_dirs:
+                        staging_dirs.append(staging_dir)
+                    files.extend(output_files)
```

Comment:
I understand why this is needed but I think this makes this go from "oof dangerous" to... "oh boy, that's a lot more potential folders"

If we're trying to detect any generated output file from a rop node - this may potentially be "anywhere" depending on whatever file type we're exporting. This now suddenly goes from "just happening to be the ROP node's output path to a lot of potential folders".

I think what I may be "roughly" ok with is if here we check whether the folder during Collect already exists or not, if not - then set it up for cleanup. (Note however that that may skip folders that may already have been created by another collector or publishing logic before this). Anyway, by only configuring folders that do not exist yet then this may just be 'cleaning up' folders that (likley) got generated due to the publishing logic itself which seems far more reasonable?

Reply:
> I think what I may be "roughly" ok with is if here we check whether the folder during Collect already exists or not, if not - then set it up for cleanup.

Directories added to 'cleanupEmptyDir' will only be cleared if they are empty. 
https://github.com/ynput/ayon-core/blob/10adfbab0e00569e814e4c9867a40f9fd6078910/client/ayon_core/plugins/publish/cleanup_explicit.py#L14-L16

Reply:
I added a note about it here https://github.com/ynput/ayon-core/pull/9/commits/d34ccc1495f742a88f783c4b702fb0ec2ca32811

Also, I add a sub folder with some files inside and the plugin was able to detect that didn't delete my directory

from `CollectFilesForCleaningUp`'s log
```
DEBUG: Add directories to 'cleanupEmptyDir': ['//storage/work/ayon_projects/Robo/Assets/Character/roboangry/work/cfx/pyblish/renders/mantra_ropMain']
```
from `ExplicitCleanUp`'s log
```
DEBUG: Skipped directories because they contain files:
\\storage\work\ayon_projects\Robo\Assets\Character\roboangry\work\cfx\pyblish\renders\mantra_ropMain
```

Reply:
> For reference, Directories added to 'cleanupEmptyDir' will only be cleared if they are empty.

I understand - what I meant is - that if for whatever reason the folder did exist before the publish.. maybe it's part of a studio's template. Then suddenly after publish it is gone because it was empty.

We could e.g. by checking before extraction just whether the folder we are considering to remove is actually not there yet only delete the empty folders we expected to have created. Anyway, maybe it's a non-issue or something for a separate issue.